services/wallet/connection/status_notifier.go

```diff
@@ -13,21 +14,22 @@ import (
 type StatusNotification map[string]State // id -> State
 
 type StatusNotifier struct {
-	statuses  map[string]*Status // id -> Status
+	statuses  *sync.Map // id -> Status
 	eventType walletevent.EventType
 	feed      *event.Feed
 }
 
-func NewStatusNotifier(statuses map[string]*Status, eventType walletevent.EventType, feed *event.Feed) *StatusNotifier {
+func NewStatusNotifier(statuses *sync.Map, eventType walletevent.EventType, feed *event.Feed) *StatusNotifier {
 	n := StatusNotifier{
 		statuses:  statuses,
 		eventType: eventType,
 		feed:      feed,
 	}
 
-	for _, status := range statuses {
-		status.SetStateChangeCb(n.notify)
-	}
+	statuses.Range(func(_, value interface{}) bool {
```

Comment:
I guess we could add `start`, `stop` and map update functions now that we're allowing changes, instead of having to rebuild the whole notifier... 
Not in this PR though, just potential improvements for the future.